File: backend/src/main/java/com/rapidphoto/features/gallery/api/dto/PhotoResponse.java
package com.rapidphoto.features.gallery.api.dto;

import com.rapidphoto.domain.PhotoStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;
import java.util.UUID;

/**
 * Detailed photo response with all metadata and versions.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PhotoResponse {

    private UUID id;

    private String fileName;

    private PhotoStatus status;

    private Long fileSize;

    private String mimeType;

    private Integer width;

    private Integer height;

    private String originalUrl;

    private String thumbnailUrl;

    private List<PhotoVersionDto> versions;

    private List<PhotoLabelDto> labels;

    private Instant createdAt;

    private Instant processedAt;

    private Instant takenAt;

    private String cameraMake;

    private String cameraModel;

    private BigDecimal gpsLatitude;

    private BigDecimal gpsLongitude;
}

File: mobile/lib/features/gallery/models/photo_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'photo_label.dart';
import 'photo_status.dart';
import 'photo_version.dart';

part 'photo_response.freezed.dart';
part 'photo_response.g.dart';

/// Detailed photo response with all metadata and versions
@freezed
abstract class PhotoResponse with _$PhotoResponse {
  

  const factory PhotoResponse({
    required String id,
    required String fileName,
    required PhotoStatus status,
    int? fileSize,
    String? mimeType,
    int? width,
    int? height,
    String? originalUrl,
    String? thumbnailUrl,
    @Default([]) List<PhotoVersion> versions,
    @Default([]) List<PhotoLabel> labels,
    @JsonKey(fromJson: _dateTimeFromEpochSeconds) required DateTime createdAt,
    @JsonKey(fromJson: _dateTimeFromEpochSecondsNullable) DateTime? processedAt,
    @JsonKey(fromJson: _dateTimeFromEpochSecondsNullable) DateTime? takenAt,
    String? cameraMake,
    String? cameraModel,
    double? gpsLatitude,
    double? gpsLongitude,
  }) = _PhotoResponse;

  factory PhotoResponse.fromJson(Map<String, dynamic> json) =>
      _$PhotoResponseFromJson(json);
}

/// Convert Unix epoch seconds (as number) to DateTime
DateTime _dateTimeFromEpochSeconds(dynamic value) {
  if (value is num) {
    return DateTime.fromMillisecondsSinceEpoch((value * 1000).round());
  }
  if (value is String) {
    return DateTime.parse(value);
  }
  throw ArgumentError('Invalid date format: $value');
}

/// Convert Unix epoch seconds (as number) to nullable DateTime
DateTime? _dateTimeFromEpochSecondsNullable(dynamic value) {
  if (value == null) return null;
  return _dateTimeFromEpochSeconds(value);
}
